enhancements/machine-api/short-circuiting-backoff.md

```diff
@@ -0,0 +1,165 @@
+create a new one. This isn't the best remediation strategy in all environments.
+
+Any Machine that enters the `Failed` state is remediated immediately, without waiting, by the MHC
+When this occurs, if the error which caused the failure is persistent (spot price too low, configuration error), replacement Machines will also be `Failed`
+As replacement machines start and fail, MHC causes a hot loop of Machine being deleted and recreated.
+Hot loop makes it difficult for users to find out why their Machines are failing.
+Another side effect of machines constantly failing, is the risk of hitting the benchmark of machine failures percentage - thus triggering the "short-circuit" mechanism which will prevent all remediations.
+
+With this enhancement we propose a better mechanism.
+In case a machine enters the `Failed` state and does not have a NodeRef or a ProviderID it'll be remediated after a certain time period has passed - thus allowing a manual intervention in order to break to hot loop.
+
+## Motivation
+
+- Preventing remediation hot loop, in order to allow a manual fix and prevent unnecessary resource usage.
+
+### Goals
+
+- Create the ability to define customized remediation for Machine that enters the `Failed` state.
+
+### Non-Goals
+
+TBD
+
+## Proposal
+
+We propose modifying the MachineHealthCheck CRD to support a FailedNodeStartupTimeout, this is the time period which controls remediation of a machine that enters the `Failed` state.
+
+### User Stories
+
+#### Story 1
+
+As an admin of a hardware based cluster, I would like failed machines to delay before automatically re-provisioning so I'll have a time frame in which to manually analyze and fix them .
+
+### Implementation Details/Notes/Constraints
+
+If no value for failedNodeStartupTimeout is defined for the MachineHealthCheck CR, the existing remediation flow
+is preserved.
+
+In case a machine enters the `Failed` state and does not have a NodeRef or a ProviderID it's remediation will be requeued by failedNodeStartupTimeout.
+After that time has passed if the machine current state remains remediation will be performed.
+
+
+#### MHC struct enhancement
+
+```go
+    type MachineHealthCheckSpec struct {
+        ...
+    
+        // +optional
+		FailedNodeStartupTimeout metav1.Duration `json:"failedNodeStartupTimeout,omitempty"`
+    }
+```
+
+#### Example CRs
+
+MachineHealthCheck:
+```yaml
+    kind: MachineHealthCheck
+    apiVersion: machine.openshift.io/v1beta1
+    metadata:
+      name: REMEDIATION_GROUP
+      namespace: NAMESPACE_OF_UNHEALTHY_MACHINE
+    spec:
+      selector:
+        matchLabels: 
+          ...
+      failedNodeStartupTimeout: 48h
+```
+
+### Risks and Mitigations
+
+No known risks
+
+## Design Details
+
+### Open Questions
+
+See deprecation and upgrade
+
+### Test Plan
+
+The existing remediation tests will be reviewed / adapted / extended as needed.
```

Comment:
i think the main point here is that we will need to add e2e tests to ensure this behaves as we expect.

Reply:
I agree that e2e will have value here, but I think UT are also important here.
IMO with this comment we allow enough flexibility without putting a focus on a specific testing phase.

Reply:
Worth adding some comments about how you might add an E2E test here

Reply:
Would still like to see some details about the testing we will be able to add here

Reply:
Hi, I assume you are not looking for info on Unit Test (which were added) and aiming to E2E tests.
I'm not that familiar with MAO e2e so I don't feel like there is more I can contribute here.
In case there is something specific you had in mind, I'll be happy to include it.